skyportal/handlers/source.py

```diff
@@ -111,10 +111,14 @@ def post(self):
         """
         data = self.get_json()
         schema = Source.__schema__()
+        user_group_ids = [g.id for g in self.current_user.groups]
+        if not user_group_ids:
+            return self.error("You must belong to one or more groups before "
+                              "you can add sources.")
         try:
-            group_ids = data.pop('group_ids')
+            group_ids = [id for id in data.pop('group_ids') if id in user_group_ids]
```

Comment:
What happens when `group_ids` is empty?

If the group is added to all user groups by default, we should document that.

Reply:
Okay -- document that where?  In the docstring description?

Reply:
Do you mean what happens if a user specifies `group_ids = []` in the request data?  In that case, it returns with an error message.

Reply:
No, more the situation where the user specifies group IDs that either don't exist or where the user doesn't have permission to post to, so that we end up with no valid IDs.

Documentation should go with the endpoint docs, so it appears in our API docs.

Reply:
In that case, we return with the error message "Invalid group_ids field. Please specify at least one valid group ID that you belong to."